ee/localserver/krypto-ec-middleware.go

```diff
@@ -137,3 +128,38 @@ func (e *kryptoEcMiddleware) Wrap(next http.Handler) http.Handler {
 		}
 	})
 }
+
+func extractChallenge(r *http.Request) (*challenge.OuterChallenge, error) {
+	// first check query parmeters
+	rawBox := r.URL.Query().Get("box")
+	if rawBox != "" {
+		decoded, err := base64.StdEncoding.DecodeString(rawBox)
+		if err != nil {
+			return nil, fmt.Errorf("decoding b64 box from url param: %w", err)
+		}
+
+		return challenge.UnmarshalChallenge(decoded)
+	}
+
+	// now check body
+	if r.Body == nil {
+		return nil, fmt.Errorf("no box found in url params or request body: body nil")
+	}
+
+	var body map[string]string
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		return nil, fmt.Errorf("unmarshalling request body to json: %w", err)
+	}
+
+	val, ok := body["box"]
```

Comment:
not sure about this .... 

was thinking that if the url param was `box=VALUE`
the body ought to be `{"box":"VALUE"}`

but the body could just simply be `VALUE`

all tangled in my head .... please comment =)

Reply:
I had exactly the same thought... "I was thinking `{"box":"VALUE"}`, but I guess it could just be the whole body"

I suspect `{"box":"VALUE"}` gives us more rope in terms of additions/changes